pisces/data_sets.py

```diff
@@ -417,28 +425,38 @@ def psg_to_sleep_wake(psg: pl.DataFrame) -> np.ndarray:
     """
     return np.where(psg[:, 1] > 0, 1, psg[:, 1])
 
-def to_WLDM(x: float) -> int:
+def to_WLDM(x: float, N4: bool=True) -> int:
     if x < 0:
         return -1
     if x == 0:
         return 0
     if x < 3:
         return 1
-    if x < 5:
-        return 2
-    return 3
+    if N4:
```

Comment:
the more I read this nested `if`, with default return, which is again repeated in the `else`, the more I think we should do a different pattern to smooth this out.

What if we instead did:
```python
def to_WLDM(x: float, N4: bool=True) -> int:
    if x < 0:
        return -1
    if x == 0:
        return 0
    if x < 3:
        return 1
    rem_value = 5 if N4 else 4
    if x < rem_value:
        return 2
    return 3
```

Reply:
Yes definitely, I agree that we need to smooth this out. Sounds good! Also there might be some overlap between my custom input/output and this PR so let's discuss how to deal with both PRs